src/seedu/addressbook/data/person/Contact.java

```diff
@@ -0,0 +1,45 @@
+package seedu.addressbook.data.person;
+
+import seedu.addressbook.data.exception.IllegalValueException;
+
+public class Contact {
+    public final String value;
+    private boolean isPrivate;
+
+    public Contact(String name, boolean isPrivate) {
+        this.value = name.trim();
+        this.isPrivate = isPrivate;
+    }
+
+    /**
+     * Validates value given a regex
+     *
+     * @throws IllegalValueException if value is invalid.
+     */
+    public void checkValid(String regex, String message) throws IllegalValueException{
+        if (!value.matches(regex)){
+            throw new IllegalValueException(message);
+        }
+    }
+
+    @Override
+    public String toString() {
+        return value;
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        return other == this // short circuit if same object
+                || (other instanceof Contact // instanceof handles nulls
+                && this.value.equals(((Contact) other).value)); // state check
```

Comment:
Note that now an Email and Address having the same value will be considered equal even if they are of different types.